tests/integration/issues/github_4555/test_port.py

```diff
@@ -0,0 +1,21 @@
+import time
+from threading import Thread
+
+import pytest
+
+from jina import Flow
+from jina.excepts import PortAlreadyUsed
+
+
+def test_two_flow_using_on_same_port_failing(port_generator):
```

Comment:
Please also add a test where you start `f1` first, build Flow `f2` (this is supposed to work!) and expect a failure on starting `f2`